### ISSUE TITLE
[Fanboy's Social] uBlock twitter share button issue

### ISSUE THREAD

**Issue by jtblin** [uBlock-LLC/uBlock#724]
I use and love uBlock, well done guys!

That said I have a few issues with the way uBlock blocks the twitter share button. Not only it hides the button but in case the button is in a list, all the items of the list! I have a few other items in the list like Github badges and they all get hidden.

![image](https://cloud.githubusercontent.com/assets/1388974/6095147/b1866f62-afa1-11e4-8a15-69b5203efdbc.png)

I personally think the twitter share button is fine and I think it's a bit annoying that people using uBlock will not see it, why blocking it? There's not many ways to do a bit of free self promotion for OSS projects ;)

Moreover, I am not able to unblock it by whitelisting it. If I had _twitter.com_ or _platform.twitter.com_, it is still blocked. Looking at the uBlock logs, the highlighted part seems to indicate that _tweet_button_ is what's blocked but that doesn't seem to be a valid entry in the whitelist.

TL;DR;
1. Do we really need to block social sites widgets by default? That seems to come from the "Fanboy’s Social Blocking List‎ (adblockplus.org)". Could that not be not enabled by default?
1. Can we avoid hiding the entire list in case one of the items need to be disabled?
1. How can the twitter button be whitelisted or is the only way to disable the entire "Fanboy's Social Blocking List"? (although that would work only for me)

**Comment by ghost** [comment]
@jtblin The whole point of using a Social Blocking List is because those who enable the list don't want to load buttons from social networks (including Twitter) around the web.

I believe Fanboy’s Social Blocking List‎ is no longer enabled by default. I uninstalled uBlock and then re-installed it from Chrome and it wasn't enabled by default. Haven't tried with the Firefox version yet.

**Comment by gorhill** [comment]
> Do we really need to block social sites widgets by default?

_"Fanboy’s Social"_ is not selected by default since [version 0.8.2.8](https://github.com/gorhill/uBlock/releases/tag/0.8.2.8).

> Can we avoid hiding the entire list in case one of the items need to be disabled?

No. Cosmetic filters are simply CSS selector, they are applied as is.

> How can the twitter button be whitelisted?

You need an exception cosmetic filter. It depends what filter you are trying to counter, and on which site. I can't answer without a specific case.

**Comment by jtblin** [comment]
Hey, thanks for the info. New version does not have the list by default indeed, but I had to uninstall and reinstall it to get rid of it, just updating keep it ticked.

On a different note, I had to disable uBlock today as it was preventing me from booking a flight on expedia.com.au (was preventing a loader lightbox to show which was really inconvenient. Feel like uBlock might be a bit too aggressive in blocking genuine stuff.

**Comment by gorhill** [comment]
@jtblin uBlock uses third-party filter lists, if you think they block too much, report to the list maintainers. You can confirm the problem is the filter lists by trying ABP with exactly the same filter lists (EasyList + EasyPrivacy). If you get the same issue, the problem is in the lists.
